src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php

```diff
@@ -0,0 +1,252 @@
+     *
+     * @return \Memcached
+     */
+    public function setupConnection(array $servers = array(), array $options = array())
+    {
+        foreach ($servers as $server) {
+            $this->addServer($server);
+        }
+
+        foreach ($options as $opt => $val) {
+            $this->setOption($opt, $val);
+        }
+
+        return $this->memcached;
+    }
+
+    /**
+     * @param mixed[] $server
+     */
+    public function addServer(array $server)
+    {
+        list($host, $port, $weight) = $this->filterServerValues(
+            isset($server[0]) ? $server[0] : self::DEFAULT_HOST,
+            isset($server[1]) ? $server[1] : self::DEFAULT_PORT,
+            isset($server[2]) ? $server[2] : self::DEFAULT_WEIGHT
+        );
+
+        if (!$this->isServerAlreadyEnabled($host, $port)) {
+            $this->memcached->addServer($host, $port, $weight);
+        }
+    }
+
+    /**
+     * @param mixed $opt
+     * @param mixed $val
+     *
+     * @return bool
+     */
+    public function setOption($opt, $val)
+    {
+        list($opt, $val) = $this->filterOptionValues($opt, $val);
+
+        return $this->memcached->setOption($opt, $val)
+            && $this->memcached->getResultCode() === \Memcached::RES_SUCCESS;
+    }
+
+    /**
+     * @param mixed $opt
+     * @param mixed $val
+     *
+     * @return mixed[]
+     */
+    private function filterOptionValues($opt, $val)
+    {
+        list($opt, $val) = array(
+            $opt = $this->tryResolvingConstant($opt),
+            $val = $this->tryResolvingConstant($val),
+        );
+
+        if (false === filter_var($opt, FILTER_VALIDATE_INT)) {
+            throw new InvalidArgumentException(sprintf('Option type invalid "%s": must resolve as int', $opt));
+        }
+
+        if (false === filter_var($val, FILTER_VALIDATE_INT) && false === filter_var($val, FILTER_VALIDATE_BOOLEAN)) {
+            throw new InvalidArgumentException(sprintf('Option value invalid "%s": must resolve as int or bool', $val));
+        }
+
+        return array($opt, $val);
+    }
+
+    /**
+     * @param mixed $val
+     *
+     * @return mixed
+     */
+    private function tryResolvingConstant($val)
+    {
+        return defined($constant = '\Memcached::'.$val) ? constant($constant) : $val;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFetch(array $ids)
+    {
+        foreach ($this->memcached->getMulti($ids) as $id => $val) {
+            try {
+                yield $id => parent::unserialize($val);
+            } catch (\Exception $e) {
+                continue;
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doHave($id)
+    {
+        return $this->memcached->get($id) !== false
+            && $this->memcached->getResultCode() !== \Memcached::RES_NOTFOUND;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doClear($namespace)
+    {
+        return $this->memcached->flush()
```

Comment:
Yeah, so I've been looking into this since I wrote this initial adapter a few weeks back. The issue revolves around the [getAllKeys()](http://php.net/manual/en/memcached.getallkeys.php) method, which would easily allow us to iterate over the keys and decide what to delete......except that method returns false instead of working properly on 80% of the configurations I've tested on. Apparently you need very specific cercumstances to come together for it to work properly. Should I implement that and then fallback to the global flush?

Also, the article you linked to is the `memcache` extension (notice it is missing the `d` --- its a different extension). *This extension, `memcached`, doesn't offer the same ability.* **But**, I would always *try to use that method* if the user also has the `memcache` extension installed?

Thoughts?